### PR TITLE
Fix warden spawn reason from DEFAULT to NATURAL

### LINKED ISSUE #8283: Warden spawned by shrieker uses DEFAULT SpawnReason
### Expected behavior

with 80+ wardens in 1 spot no other mobs should be spawning

### Observed/Actual behavior

no other mobs should be spawning

### Steps/models to reproduce

get 80 wardens spawned look around mobs there is

### Plugin and Datapack List

image

### Paper version

image

### Other

_No response_

### PR DESCRIPTION
**Pull request by Kenuux**
This fix is trivial. fixes #8283

### CODE REVIEW

File: patches/server/0958-Fix-warden-spawn-reason-from-DEFAULT-to-NATURAL.patch [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+diff --git a/src/main/java/net/minecraft/world/level/block/entity/SculkShriekerBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/SculkShriekerBlockEntity.java
+index 0554afa72ae6d2f01886916031c8b90b0d45eb52..a96f66e7a79078163214409c29709b0e331482cb 100644
+--- a/src/main/java/net/minecraft/world/level/block/entity/SculkShriekerBlockEntity.java
++++ b/src/main/java/net/minecraft/world/level/block/entity/SculkShriekerBlockEntity.java
+@@ -36,6 +36,7 @@ import net.minecraft.world.level.gameevent.GameEvent;
+ import net.minecraft.world.level.gameevent.GameEventListener;
+ import net.minecraft.world.level.gameevent.vibrations.VibrationListener;
+ import net.minecraft.world.phys.Vec3;
++import org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason;
```
Comment (NoahvdAa) [inline patches/server/0958-Fix-warden-spawn-reason-from-DEFAULT-to-NATURAL.patch]: Use the fully qualified name instead of adding an import

**Review by NoahvdAa — COMMENTED** [review]
👋 Welcome to Paper, congrats on making your first PR!

File: patches/server/0958-Fix-warden-spawn-reason-from-DEFAULT-to-NATURAL.patch (hunk 1 above)
Comment (Kenuux) [reply patches/server/0958-Fix-warden-spawn-reason-from-DEFAULT-to-NATURAL.patch]: Done

**Comment by Doc94** [comment]
Looks good, maybe this change can be included in [server/0939-Fixes-and-additions-to-the-SpawnReason-API.patch](https://github.com/PaperMC/Paper/blob/master/patches/server/0939-Fixes-and-additions-to-the-SpawnReason-API.patch) ?

**Comment by electronicboy** [comment]
Natural would imply that the mob was spawned by the natural mob spawning logic, that doesn't look to be the case here

The original issue appears to be WAI outside of the spawn reason shouldn't be default, but, I don't think that it should be "natural" either. There is maybe some wider issue around what all counts towards the limit, the years of butchering to the mob spawning and persistence logic really creates a lot of messes around how mob spawn limits work

**Comment by Kenuux** [comment]
> Natural would imply that the mob was spawned by the natural mob spawning logic, that doesn't look to be the case here
> 
> The original issue appears to be WAI outside of the spawn reason shouldn't be default, but, I don't think that it should be "natural" either. There is maybe some wider issue around what all counts towards the limit, the years of butchering to the mob spawning and persistence logic really creates a lot of messes around how mob spawn limits work

Yeah, I thought so too. But then I saw that there used to be this pull request (https://github.com/PaperMC/Paper/pull/8296) that was closed by the author for no reason. Based on the comments, I assumed that we just take NATURAL now.

**Review by Machine-Maker — APPROVED** [review]

**Pull request merged**